galsim/correlatednoise.py

```diff
@@ -117,6 +121,106 @@ def __init__(self, rng, gsobject, wcs=PixelScale(1.0)):
+           comparable to that in the final output, and ``cn`` has been rotated or otherwise transformed
+           (e.g.  via the `BaseCorrelatedNoise.rotate`, `BaseCorrelatedNoise.shear` methods; see below).
+
+        Conversely, the approximation will work best in the case where the correlated noise used to
+        instantiate the ``cn`` is taken from an input image for which ``image.scale`` is smaller than
+        that in the desired output.  This is the most common use case in the practical treatment of
+        correlated noise when simulating galaxies from space as observed in ground-based surveys.
+
+        Changing from the default bilinear interpolant is made possible, but not recommended.  For more
+        information please see the discussion on https://github.com/GalSim-developers/GalSim/pull/452.
+
+        Parameters:
+            file_name:      The name of the file to read.
+            rng:            If provided, a random number generator to use as the random number
+                            generator of the resulting noise object. (may be any kind of
+                            `BaseDeviate` object) [default: None, in which case, one will be
+                            automatically created, using the time as a seed.]
+            scale:          If desired, an image pixel scale to override any scale given in the file.
+            variance:       Scales the correlation function so that its point variance, equivalent
+                            to its value at zero separation distance, matches this value.
+                            [default: 0., which means to use the variance in the original file.]
+            x_interpolant:  Forces use of a non-default interpolant for interpolation of the
+                            internal lookup table in real space.  See below for more details.
+                            [default: galsim.Linear()]
+            gsparams:       An optional `GSParams` argument. [default: None]
+
+        Returns:
+            a `BaseCorrelatedNoise` instance
+        """
+        from . import fits
+        from .interpolant import Linear
+        from .interpolatedimage import InterpolatedImage
+        if not os.path.isfile(file_name):
+            raise OSError("The file %r does not exist."%(file_name))
+        try:
+            cfimage = fits.read(file_name)
+        except (OSError, AttributeError, TypeError) as e:
+            raise OSError("Unable to read file %s.\n%r"%(file_name,e))
+
+        # Check for invalid images:
+        if cfimage.nrow != cfimage.ncol:
+            raise GalSimError("Input image is not square.")
+        if cfimage.nrow % 2 != 1:
+            raise GalSimError("Input image does not have odd size.")
+        if np.max(np.abs((cfimage - cfimage.rot_180()).array)) > 1.e-12:
+            raise GalSimError("Input image does not have 180 degree rotational symmetry.")
+
+        # Also check for invalid negative variance
```

Comment:
Should we also check that the cross-covariances obey -1<=C[i,j]/C[0,0]<=1?  Am I right in thinking that this must be mathematically the case or is that not true?

Reply:
This could help avoid a scenario, for example, that somebody puts the variance at (0,0) instead of in the center.  Such a case might pass your previous test if the center pixel happens to have a positive cross-covariance, no matter how small.

Reply:
Yep, good idea.  I added that.